### PR TITLE
refactor: prevent logo click reloading homepage if already there

### PR DESCRIPTION
**Pull request by Jer-Pha**
This is only necessary if you agree with [this comment](https://ux.stackexchange.com/a/55642) about it being a design error. There's also the very small impact to bandwidth of an unnecessary page load (pretty insignificant for the site's traffic, though).

A different option is to keep the anchor tag and only adjust the href to "#".  This tells users that the logo is clickable without reloading the page.

### CODE REVIEW

File: pythonsd/templates/pythonsd/base.html [hunk 1]
```diff
@@ -23,9 +23,14 @@
...
-      <a class="navbar-brand" href="{% url 'index' %}">
-        <img src="{% static 'img/sandiegopython-logo.svg' %}" alt="San Diego Python Home">
+    {% url 'index' as index_url %}
+    {% if request.path == index_url %}
+		<img src="{% static 'img/sandiegopython-logo.svg' %}" id="navbar-logo" class="navbar-brand" alt="San Diego Python Home">
+    {% else %}
+      <a class="navbar-brand" href="{{ index_url }}">
+		<img src="{% static 'img/sandiegopython-logo.svg' %}" id="navbar-logo" alt="San Diego Python Home">
       </a>
+    {% endif %}
```
Comment (davidfischer) [inline pythonsd/templates/pythonsd/base.html]: I prefer the following to duplicating the image tag:

```suggestion
      {% url 'index' as index_url %}
      {% if request.path != index_url %}<a class="navbar-brand" href="{{ index_url }}">{% endif %}
		  <img src="{% static 'img/sandiegopython-logo.svg' %}" id="navbar-logo" class="navbar-brand" alt="San Diego Python Home">
      {% if request.path != index_url %}</a>{% endif %}
```

File: assets/src/sass/_theme.scss [hunk 2]
```diff
@@ -1,7 +1,5 @@
-.navbar-brand {
```
Comment (davidfischer) [inline assets/src/sass/_theme.scss]: I found a slight issue while testing this out. The top navbar is a slightly different size on the homepage vs. the code of conduct page after this change. `navbar-brand` has some built-in CSS associated with it from bootstrap and that's the reason why. Removing it makes the top navigation different sizes when navigating and that's much more awkward than an extra link.

Screenshot from 2024-05-23 17-40-21
Screenshot from 2024-05-23 17-40-32

File: pythonsd/templates/pythonsd/base.html (hunk 1 above)
Comment (davidfischer) [inline pythonsd/templates/pythonsd/base.html]: This is the alternative using `href="#"`. If this is accepted, please also revert the changes to `_theme.scss`:

```suggestion
      {% url 'index' as index_url %}
      <a class="navbar-brand" href="{% if request.path != index_url %}{{ index_url }}{% else %}#{% endif %}">
        <img src="{% static 'img/sandiegopython-logo.svg' %}" id="navbar-logo" alt="San Diego Python Home">
      </a>
```

**Review by davidfischer — CHANGES_REQUESTED** [review]
> This is only necessary if you agree with [this comment](https://ux.stackexchange.com/a/55642) about it being a design error. There's also the very small impact to bandwidth of an unnecessary page load (pretty insignificant for the site's traffic, though).

I don't feel particularly strongly one way or the other. I'd be fine merging this if we can work out the issues but I don't mind the current functionality either.

> A different option is to keep the anchor tag and only adjust the href to "#". This tells users that the logo is clickable without reloading the page.

Given the issues with sizing, that might be a better solution. It also simplifies the `{% if %}/{% else %}` logic.

File: assets/src/sass/_theme.scss (hunk 2 above)
Comment (Jer-Pha) [reply assets/src/sass/_theme.scss]: This is odd. I'm seeing them exactly the same. I'm not sure where that 10px is coming from but it's not worth the trouble. I'll revert the CSS changes and go with the "#" solution.

image

File: pythonsd/templates/pythonsd/base.html (hunk 1 above)
Comment (Jer-Pha) [reply pythonsd/templates/pythonsd/base.html]: If the CSS screenshot is from this code, that is where the 10px is coming from. This would have added the padding for the navbar-brand class twice, `<a>` and ``.

File: assets/src/sass/_theme.scss (hunk 2 above)
Comment (davidfischer) [reply assets/src/sass/_theme.scss]: You are right. It came from here: https://github.com/sandiegopython/pythonsd-django/pull/152#discussion_r1614130354

**Review by davidfischer — APPROVED** [review]
I'm glad we got to something we're happy with after a bit of back and forth! This looks good. :shipit:
